website/docs/cloud-docs/integrations/kubernetes/ops-v2-migration.mdx

```diff
@@ -43,33 +43,220 @@ Next, backup the workspace resources.
+        value: "s3cr3t"
+        hcl: false
+        sensitive: false
+        environmentVariable: true
+```
 
-```shell-session
-kubectl patch workspace <WORKSPACE_NAME> --type=merge --patch '{"spec": {"applyMethod": "manual"}}'
+</CodeBlockConfig>
+
+In v2 of the operator, you must configure Terraform variables in `spec.terraformVariables` and environment variables `spec.environmentVariables`.
+
+<CodeBlockConfig filename="v2">
+
+```yaml
+apiVersion: app.terraform.io/v1alpha2
+kind: Workspace
+metadata:
+  name: migration
+  spec:
+    terraformVariables:
+      - name: username
+        value: "user"
+        hcl: true
+        sensitive: false
+    environmentVariables:
+      - name: SECRET_KEY
+        value: "s3cr3t"
+        hcl: false
+        sensitive: false
 ```
-## Upgrade the operator
 
-Clone the Terraform [Cloud Kubernetes Operator repository](https://github.com/hashicorp/terraform-cloud-operator).
+</CodeBlockConfig>
 
-```shell-session
-$ git clone https://github.com/hashicorp/terraform-cloud-operator.git
+### Module controller
+
+Terraform Cloud Kubernetes Operator v2 configures modules in a new `Module` controller separate from the `Workspace` controller. Below is a template of a custom resource manifest:
+
+```yaml
+---
+apiVersion: app.terraform.io/v1alpha2
+kind: Module
+metadata:
+  name: <NAME>
+spec:
+  organization: <ORG-NAME>
+  token:
+    secretKeyRef:
+      name: <SECRET-NAME>
+      key: <KEY-NAME>
+  name: operator
 ```
 
-Change to the migration directory.
+The table below details the mapping between the `Workspace` controller in v1 of the operator and the `Module` controller in v2 of the operator.
+
+| Version 1 (Workspace CRD) | Version 2 (Module CRD) | Notes |
+| --- | --- | --- |
+| `spec.module` | N/A | In v2 of the operator a `Module` is a separate controller with its own CRD. |
+| N/A | `spec.name: operator` | In v1 of the operator, the name of the generated module is hardcoded to `operator`. In v2, the default name of the generated module is `this`, however it can be changed. |
+| `spec.module.source` | `spec.module.source` | This supports all Terraform [module sources](https://developer.hashicorp.com/terraform/language/modules/sources). |
+| `spec.module.version` | `spec.module.version` | Refer to  [module sources](https://developer.hashicorp.com/terraform/language/modules/sources) for versioning information for each module source. |
+| `spec.variables.[*]` | `spec.variables.[*].name` | Variable names should be included in the module. This is a reference to variables in the Workspace where the module will be executed. |
+| `spec.outputs.[*].key` | `spec.outputs.[*].name` | Output names should be included in the module. This is a reference to the output variables produced by the module. |
+| `status.workspaceID` OR `metadata.namespace-metadata.name` | `spec.workspace.id` OR `spec.workspace.name` | The workspace where the module will be executed. The workspace must be within the same organization. |
```

Comment:
In the same organization as what? Or are you referring to listing multiple workspace names and those workspaces must be in the same org?
```suggestion
| `status.workspaceID` OR `metadata.namespace-metadata.name` | `spec.workspace.id` OR `spec.workspace.name` | The workspace where the module is executed. The workspace must be in the same organization. |
```